When `ITK_WRAP_PYTHON` is turned on during the initial configure, it correctly complains about version:
```log
dzenan.zukic@krs-7714:~/ITK/ITK-py-direct$ cmake -DITK_WRAP_PYTHON=ON ../ITK-git/
-- Setting build type to 'Release' as none was specified.
-- The CXX compiler identification is GNU 11.4.0
-- The C compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /usr/bin/git (found version "2.34.1")
-- Performing Test C_HAS_WARNING-Wno-unused-parameter
-- Performing Test C_HAS_WARNING-Wno-unused-parameter - Success
-- Performing Test C_HAS_WARNING-Wall
-- Performing Test C_HAS_WARNING-Wall - Success
-- Performing Test C_HAS_WARNING-Wcast-align
-- Performing Test C_HAS_WARNING-Wcast-align - Success
-- Performing Test C_HAS_WARNING-Wdisabled-optimization
-- Performing Test C_HAS_WARNING-Wdisabled-optimization - Success
-- Performing Test C_HAS_WARNING-Wextra
-- Performing Test C_HAS_WARNING-Wextra - Success
-- Performing Test C_HAS_WARNING-Wformat_2
-- Performing Test C_HAS_WARNING-Wformat_2 - Success
-- Performing Test C_HAS_WARNING-Winvalid-pch
-- Performing Test C_HAS_WARNING-Winvalid-pch - Success
-- Performing Test C_HAS_WARNING-Wpointer-arith
-- Performing Test C_HAS_WARNING-Wpointer-arith - Success
-- Performing Test C_HAS_WARNING-Wshadow
-- Performing Test C_HAS_WARNING-Wshadow - Success
-- Performing Test C_HAS_WARNING-Wunused
-- Performing Test C_HAS_WARNING-Wunused - Success
-- Performing Test C_HAS_WARNING-Wwrite-strings
-- Performing Test C_HAS_WARNING-Wwrite-strings - Success
-- Performing Test CXX_HAS_WARNING-Wall
-- Performing Test CXX_HAS_WARNING-Wall - Success
-- Performing Test CXX_HAS_WARNING-Wcast-align
-- Performing Test CXX_HAS_WARNING-Wcast-align - Success
-- Performing Test CXX_HAS_WARNING-Wdisabled-optimization
-- Performing Test CXX_HAS_WARNING-Wdisabled-optimization - Success
-- Performing Test CXX_HAS_WARNING-Wextra
-- Performing Test CXX_HAS_WARNING-Wextra - Success
-- Performing Test CXX_HAS_WARNING-Wformat_2
-- Performing Test CXX_HAS_WARNING-Wformat_2 - Success
-- Performing Test CXX_HAS_WARNING-Winvalid-pch
-- Performing Test CXX_HAS_WARNING-Winvalid-pch - Success
-- Performing Test CXX_HAS_WARNING-Wpointer-arith
-- Performing Test CXX_HAS_WARNING-Wpointer-arith - Success
-- Performing Test CXX_HAS_WARNING-Wshadow
-- Performing Test CXX_HAS_WARNING-Wshadow - Success
-- Performing Test CXX_HAS_WARNING-Wunused
-- Performing Test CXX_HAS_WARNING-Wunused - Success
-- Performing Test CXX_HAS_WARNING-Wwrite-strings
-- Performing Test CXX_HAS_WARNING-Wwrite-strings - Success
-- Performing Test CXX_HAS_WARNING-Wno-undefined-var-template
-- Performing Test CXX_HAS_WARNING-Wno-undefined-var-template - Failed
-- Performing Test CXX_HAS_WARNING-Woverloaded-virtual
-- Performing Test CXX_HAS_WARNING-Woverloaded-virtual - Success
-- Performing Test CXX_HAS_WARNING-Wctad-maybe-unsupported
-- Performing Test CXX_HAS_WARNING-Wctad-maybe-unsupported - Success
-- Performing Test CXX_HAS_WARNING-Wstrict-null-sentinel
-- Performing Test CXX_HAS_WARNING-Wstrict-null-sentinel - Success
-- Performing Test CXX_HAS_DISABLE_OPTIMIZATION_FLAG
-- Performing Test CXX_HAS_DISABLE_OPTIMIZATION_FLAG - Failed
-- Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found suitable version "3.13.2", required range is "3.11...3.999")
-- Could NOT find Python3 (missing: Python3_NumPy_INCLUDE_DIRS NumPy) (found suitable version "3.13.2", minimum required is "3.13.2")
-- Python3_FOUND=FALSE
--  o Python3 Found COMPONENT: Python3_Interpreter_FOUND: TRUE
--  o Python3 Found COMPONENT: Python3_Development.Module_FOUND: TRUE
--  o Python3 Found COMPONENT: Python3_Development.SABIModule_FOUND: TRUE
--  o Python3 Missing COMPONENT: Python3_NumPy_FOUND: FALSE
CMake Error at CMake/ITKSetPython3Vars.cmake:178 (message):
  At least 1 required Python3 COMPONENT could not be found from :
  Interpreter;Development.Module;Development.SABIModule;NumPy

            in range 3.11...3.999:
            Python3_EXECUTABLE=:/home/local/KHQ/dzenan.zukic/miniconda3/bin/python3.13:
            ITK_WRAP_PYTHON_VERSION=:3.13.2:
            Python3_ROOT_DIR=::
            ---
            Python3_FOUND=FALSE
            Python3_Interpreter_FOUND=TRUE
            Python3_Compiler_FOUND=FALSE
            Python3_Development_FOUND=FALSE
            Python3_Development.Module_FOUND=TRUE
            Python3_Development.SABIModule_FOUND=TRUE
            Python3_Development.Embed_FOUND=FALSE
            Python3_NumPy_FOUND=FALSE
      
Call Stack (most recent call first):
  CMakeLists.txt:758 (include)


-- Configuring incomplete, errors occurred!
```

--- a/CMake/ITKSetPython3Vars.cmake
+++ b/CMake/ITKSetPython3Vars.cmake
@@ -168,6 +168,7 @@ else()
     list(APPEND _python_find_components NumPy)
   endif()
   set(_missing_required_component FALSE)
+  set(_missing_python_components "")
   find_package(
     Python3
     ${ITK_WRAP_PYTHON_VERSION}
@@ -183,6 +184,7 @@ else()
         " o Python3 Missing COMPONENT: Python3_${_required_component}_FOUND: ${Python3_${_required_component}_FOUND}"
       )
       set(_missing_required_component TRUE)
+      list(APPEND _missing_python_components ${_required_component})
     else()
       message(
         STATUS
@@ -192,9 +194,39 @@ else()
   endforeach()
   unset(_required_component)
   if(_missing_required_component)
+    set(_missing_component_remedy "")
+    if(NumPy IN_LIST _missing_python_components)
+      string(
+        APPEND
+        _missing_component_remedy
+        "
+          NumPy: ${Python3_EXECUTABLE} -m pip install numpy"
+      )
+    endif()
+    if(
+      Development.Module
+        IN_LIST
+        _missing_python_components
+      OR
+        Development.SABIModule
+          IN_LIST
+          _missing_python_components
+    )
+      string(
+        APPEND
+        _missing_component_remedy
+        "
+          Development: install the development headers matching this interpreter
+                       (Debian/Ubuntu: apt install python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}-dev)"
+      )
+    endif()
     message(
       FATAL_ERROR
-      "At least 1 required Python3 COMPONENT could not be found from : ${_python_find_components}
+      "Missing required Python3 COMPONENT(s): ${_missing_python_components}${_missing_component_remedy}
+
+          Or set ITK_WRAP_PYTHON=OFF to build without Python wrapping.
+          ---
+          Searched for : ${_python_find_components}
           in range ${PYTHON_VERSION_MIN}...${PYTHON_VERSION_MAX}:
           Python3_EXECUTABLE=:${Python3_EXECUTABLE}:
           ITK_WRAP_PYTHON_VERSION=:${ITK_WRAP_PYTHON_VERSION}:
@@ -210,12 +242,14 @@ else()
           Python3_NumPy_FOUND=${Python3_NumPy_FOUND}
     "
     )
+    unset(_missing_component_remedy)
   else()
     message(STATUS " o Python3_EXECUTABLE=${Python3_EXECUTABLE}")
     message(STATUS " o Python3_ROOT_DIR=${Python3_ROOT_DIR}")
     message(STATUS " o ITK_WRAP_PYTHON_VERSION=${ITK_WRAP_PYTHON_VERSION}")
   endif()
   unset(_missing_required_component)
+  unset(_missing_python_components)
   unset(_python_find_components)
   # _ITK_MINIMUM_SUPPORTED_LIMITED_API_VERSION is cached (INTERNAL) and intentionally
   # left set so itk_end_wrap_module.cmake can pin USE_SABI to the same floor.

--- a/CMake/ITKSetPython3Vars.cmake
+++ b/CMake/ITKSetPython3Vars.cmake
@@ -7,7 +7,9 @@
 # may become less reliable with newer versions of CMake (as opposed setting FindPython3 HINTS).  Current
 # implementation gives preference to active virtualenvs.
 cmake_policy(SET CMP0094 NEW) # makes FindPython3 prefer activated virtualenv Python to latest version
-set(PYTHON_VERSION_MIN 3.11)
+# Canonical floor; a specified Python3_EXECUTABLE narrows PYTHON_VERSION_MIN below, so keep it separate.
+set(ITK_WRAP_PYTHON_MINIMUM_VERSION 3.11)
+set(PYTHON_VERSION_MIN ${ITK_WRAP_PYTHON_MINIMUM_VERSION})
 set(PYTHON_VERSION_MAX 3.999)
 if(DEFINED Python3_EXECUTABLE) # if already specified
   set(_specified_Python3_EXECUTABLE ${Python3_EXECUTABLE})
@@ -22,6 +24,22 @@ if(DEFINED Python3_EXECUTABLE) # if already specified
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
   if(_specified_Python3_VERSION_MM)
+    # execute_process re-runs every configure, so this also rejects incremental
+    # reconfigures of a cached below-floor tree, before any wrapping work.
+    if(
+      PYTHON_DEVELOPMENT_REQUIRED
+      AND
+        _specified_Python3_VERSION_MM
+          VERSION_LESS
+          ${ITK_WRAP_PYTHON_MINIMUM_VERSION}
+    )
+      message(
+        FATAL_ERROR
+        "ITK Python wrapping (ITK_WRAP_PYTHON=ON) requires Python >= ${ITK_WRAP_PYTHON_MINIMUM_VERSION}, "
+        "but the specified Python3_EXECUTABLE=${_specified_Python3_EXECUTABLE} is Python ${_specified_Python3_VERSION_MM}. "
+        "Provide a Python >= ${ITK_WRAP_PYTHON_MINIMUM_VERSION} interpreter, or set ITK_WRAP_PYTHON=OFF."
+      )
+    endif()
     set(PYTHON_VERSION_MIN ${_specified_Python3_VERSION_MM})
     set(PYTHON_VERSION_MAX ${_specified_Python3_VERSION_MM})
   endif()
@@ -56,6 +74,25 @@ else()
       NumPy
   )
   set(ITK_WRAP_PYTHON_VERSION "${Python3_VERSION}")
+
+  # An empty Python3_VERSION means the range found nothing; otherwise the second
+  # find_package below runs unversioned and silently accepts a below-floor Python.
+  if(
+    NOT
+      Python3_VERSION
+    OR
+      Python3_VERSION
+        VERSION_LESS
+        ${ITK_WRAP_PYTHON_MINIMUM_VERSION}
+  )
+    message(
+      FATAL_ERROR
+      "ITK Python wrapping (ITK_WRAP_PYTHON=ON) requires Python >= ${ITK_WRAP_PYTHON_MINIMUM_VERSION}, "
+      "but no such interpreter was found in range ${PYTHON_VERSION_MIN}...${PYTHON_VERSION_MAX} "
+      "(resolved version '${Python3_VERSION}', Python3_EXECUTABLE='${Python3_EXECUTABLE}'). "
+      "Provide a Python >= ${ITK_WRAP_PYTHON_MINIMUM_VERSION} interpreter, or set ITK_WRAP_PYTHON=OFF."
+    )
+  endif()
 
   # start section to define package components based on LIMITED_API support and choices
   # Cached so the abi3 floor is a single source of truth shared with the module

